src/renderers/common/Buffer.js

```diff
@@ -44,6 +44,46 @@ class Buffer extends Binding {
 		 */
 		this._buffer = buffer;
 
+		/**
+		 * An array of update ranges.
+		 *
+		 * @private
+		 * @type {Array<{start: number, count: number}>}
+		 */
+		this._updateRanges = [];
```

Comment:
Why are you using a private property here? If you make it public, you won't need the getter.

Reply:
I need the Node Array reference in `NodeUniformBuffer` when used in `BufferNode`; it would be safer to override it than to assign it in the declaration.

https://github.com/sunag/three.js/blob/7b0d792bb0a39092947b01d686eb11539ae0b447/src/renderers/common/nodes/NodeUniformBuffer.js#L60-L70